src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -79,5 +83,193 @@ public string VersionString
+        /// </summary>
+        public static bool IsFreeBSDVersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)
+            => IsFreeBSD() && IsOSVersionAtLeast(major, minor, build, revision);
+
+        /// <summary>
+        /// Indicates whether the current application is running on Android.
+        /// </summary>
+        public static bool IsAndroid() =>
+#if TARGET_ANDROID
+            true;
+#else
+            false;
+#endif
+
+        /// <summary>
+        /// Check for the Android version (returned by 'uname') with a >= version comparison. Used to guard APIs that were added in the given Android release.
+        /// </summary>
+        public static bool IsAndroidVersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)
+            => IsAndroid() && IsOSVersionAtLeast(major, minor, build, revision);
+
+        /// <summary>
+        /// Indicates whether the current application is running on iOS.
+        /// </summary>
+        public static bool IsIOS() =>
+#if TARGET_IOS
+            true;
+#else
+            false;
+#endif
+
+        /// <summary>
+        /// Check for the iOS version (returned by 'libobjc.get_operatingSystemVersion') with a >= version comparison. Used to guard APIs that were added in the given iOS release.
+        /// </summary>
+        public static bool IsIOSVersionAtLeast(int major, int minor = 0, int build = 0)
+            => IsIOS() && IsOSVersionAtLeast(major, minor, build, 0);
+
+        /// <summary>
+        /// Indicates whether the current application is running on macOS.
+        /// </summary>
+        public static bool IsMacOS() =>
+#if TARGET_OSX
+            true;
+#else
+            false;
+#endif
+
+        /// <summary>
+        /// Check for the macOS version (returned by 'libobjc.get_operatingSystemVersion') with a >= version comparison. Used to guard APIs that were added in the given macOS release.
+        /// </summary>
+        public static bool IsMacOSVersionAtLeast(int major, int minor = 0, int build = 0)
+            => IsMacOS() && IsOSVersionAtLeast(major, minor, build, 0);
+
+        /// <summary>
+        /// Indicates whether the current application is running on tvOS.
+        /// </summary>
+        public static bool IsTvOS() =>
+#if TARGET_TVOS
+            true;
+#else
+            false;
+#endif
+
+        /// <summary>
+        /// Check for the tvOS version (returned by 'libobjc.get_operatingSystemVersion') with a >= version comparison. Used to guard APIs that were added in the given tvOS release.
+        /// </summary>
+        public static bool IsTvOSVersionAtLeast(int major, int minor = 0, int build = 0)
+            => IsTvOS() && IsOSVersionAtLeast(major, minor, build, 0);
+
+        /// <summary>
+        /// Indicates whether the current application is running on watchOS.
+        /// </summary>
+        public static bool IsWatchOS() =>
+#if TARGET_WATCHOS
+            true;
+#else
+            false;
+#endif
```

Comment:
@akoeplinger @marek-safar I know that this question might be silly, but do we officially support WatchOS as part of .NET 5? I could not find anything like `<TargetsWatchOS>` or `TARGET_WATCHOS` except for Mono-specific `.c` and `.h` [files](https://github.com/dotnet/runtime/search?p=1&q=TARGET_WATCHOS&unscoped_q=TARGET_WATCHOS). 

it would mean that we are adding the public contract of this method now, but it would always return `false`?

Reply:
> it would mean that we are adding the public contract of this method now, but it would always return `false`?

Seems fine to me. Still useful for libraries built against .NET 5.

Reply:
> Seems fine to me. Still useful for libraries built against .NET 5.

@terrajobst  Thanks for confirmation. 👍

Reply:
> do we officially support WatchOS as part of .NET 5?

We don't but I agree with Immo that could be useful for others.